### PR TITLE
[stdlib] Snake casing stdlib

### PR DESCRIPTION
**Pull request by msaelices**
Convert camelCase code to a pythonic snake_case throughout the stdlib

### CODE REVIEW

**Review by JoeLoser — APPROVED** [review]

**Comment by JoeLoser** [comment]
!sync

File: stdlib/src/sys/_assembly.mojo [hunk 1]
```diff
@@ -40,7 +40,7 @@ fn inlined_assembly[
                 _type=None,
                 assembly = asm.value,
                 constraints = constraints.value,
-                hasSideEffects = __mlir_attr.unit,
+                has_side_effects = __mlir_attr.unit,
```
Comment (zyx-billy) [inline stdlib/src/sys/_assembly.mojo:43]: hey thanks for this refactor! Unfortunately the attribute names for MLIR ops can't be changed as they need to be in sync with what the ops expect. Do you mind reverting all these `has_side_effects` name changes (in both files)? Thanks!

Comment (JoeLoser) [reply stdlib/src/sys/_assembly.mojo:43]: +1 to what Billy said here: it's the reason internal CI is failing and we haven't been able to land this change yet.

Comment (JoeLoser) [reply stdlib/src/sys/_assembly.mojo:43]: @msaelices if you rebase and take Billy's comments into account, we're happy to land this!

Comment (msaelices) [reply stdlib/src/sys/_assembly.mojo]: Done. Thanks!

**Comment by JoeLoser** [comment]
!sync

**Review by JoeLoser — APPROVED** [review]

**Comment by JoeLoser** [comment]
!sync

**Pull request closed without merging**